### PR TITLE
Bug 1838903 - Update ktlint to version 0.49.1

### PR DESCRIPTION
**Pull request by rvandermeulen**
https://bugzilla.mozilla.org/show_bug.cgi?id=1838903
https://bugzilla.mozilla.org/show_bug.cgi?id=1838903
https://bugzilla.mozilla.org/show_bug.cgi?id=1838903

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt [hunk 1]
```diff
@@ -106,9 +106,9 @@ class TrackingProtectionPolicyFactory(
     }
 }
 
+@Suppress("MaxLineLength")
 @VisibleForTesting
-internal fun TrackingProtectionPolicyForSessionTypes.applyTCPIfNeeded(settings: Settings):
-    TrackingProtectionPolicyForSessionTypes {
+internal fun TrackingProtectionPolicyForSessionTypes.applyTCPIfNeeded(settings: Settings): TrackingProtectionPolicyForSessionTypes {
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt:111]: Could we undo this change to not hit the `MaxLineLength`?

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
This will need to be rebased when https://github.com/mozilla-mobile/firefox-android/pull/2462 lands

**Review by jonalmeida — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt (hunk 1 above)
Comment (rvandermeulen) [reply fenix/app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt:111]: @gabrielluong The alternative is suppressing a `Single space expected between colon and return type (standard:function-return-type-spacing)` error as I don't see a good way of appeasing that rule without exceeding the maximum line length. IMO, I found the line length suppression to be the reasonable trade-off for better readability and consistency throughout the file. Let me know, though!

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt:111]: Sounds reasonable. I guess this is more of an unfortunate issue with the how we have named things here.
